proposals/memory-specializations/utilities.tex

```diff
@@ -0,0 +1,363 @@
+            @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>()}@;
+        @\added{return first;}@
+\end{codeblock}
+
+\rSec3[uninitialized.copy]{uninitialized_copy}
+\begin{codeblock}
+  template <@\changed{class}{InputIterator}@ @\changed{InputIterator}{I}@, @\added{Sentinel<I> S, }@
+            @\changed{class ForwardIterator}{ForwardIterator O}@>
+    @\added{requires }\added{Constructible<value_type_t<O>, reference_t<O>{>}() \&\&}@
+             @\replacedtxt{Same<value_type_t<O>\&, reference_t<O>{>} \&\&}{__ReferenceTo<O, value_type_t<O>{>}()}@
+             @\newtxt{CopyConstructible<value_type_t<O>{>}()}@
+  @\changed{ForwardIterator}{tagged_pair<tag::in(I), tag::out(O)>}@
+    uninitialized_copy(@\changed{InputIterator}{I}@ first, @\changed{InputIterator}{S}@ last, @\changed{ForwardIterator}{O}@ result);
+\end{codeblock}
+
+\setcounter{Paras}{0}
+\pnum
+\effects \changed{As if by}{Equivalent to}:
+\begin{codeblock}
+        for (; first != last; ++first)
+          ::new (static_cast<void*>(addressof(*first)))
+            @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>}@(*first);
+        @\added{return \{last, result + (last - first)\}}@
+\end{codeblock}
+
+{\color{addclr}
+\pnum
+\requires \tcode{result} shall not be in the range \tcode{[first, last)}.
+
+\begin{codeblock}
+  template <InputRange Rng, class O>
+    requires ForwardIterator<O>() &&
+             Constructible<value_type_t<O>, reference_t<iterator_t<Rng>>>() &&
+             @\replacedtxt{Same<value_type_t<O>\&, reference_t<O>{>} \&\&}{__ReferenceTo<O, value_type_t<O>{>}()}@
+             @\oldtxt{CopyConstructible<value_type_t<O>{>}()}@
+  tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>
+    uninitialized_copy(Rng&& rng, O&& result);
+\end{codeblock}
+
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        return uninitialized_copy(begin(rng), end(rng), std::forward<O>(result));
+\end{codeblock}
+\pnum
+\requires \tcode{result} shall not be in the range \tcode{[first, last)}.
+} %% \color{addclr}
+
+\begin{codeblock}
+  template <@\changed{class}{InputIterator}@ @\changed{InputIterator}{I}@, @\changed{class Size, class ForwardIterator}{OutputIterator O}@>
+    @\added{requires }\added{Constructible<value_type_t<O>, reference_t<I>{>}() \&\&}@
+             @\replacedtxt{Same<value_type_t<O>\&, reference_t<O>{>}}{__ReferenceTo<O, value_type_t<O>{>}()}\added{ \&\&}@
+             @\newtxt{CopyConstructible<value_type_t<O>{>}()}@
+  @\changed{ForwardIterator}{tagged_pair<tag::in(I), tag::out(O)>}@
+    uninitialized_copy_n(@\changed{InputIterator}{I\&\&}@ first, @\changed{Size}{difference_type_t<I>}@ n,
+                         @\changed{ForwardIterator}{O\&\&}@ out)
+\end{codeblock}
+
+\pnum
+\effects \changed{As if by}{Equivalent to:}
+\begin{codeblock}
+        for (; n > 0; ++result, (void) ++first, --n)
+          ::new(static_cast<void*>(addressof(result)))
```

Comment:
Stopping my review here. Too many mistakes. Please take a hard look at the remaining algorithms.